src/include/scope_guard.h

```diff
@@ -0,0 +1,37 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2016 Red Hat
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+#ifndef SCOPE_GUARD
+#define SCOPE_GUARD
+
+template <typename F>
+struct scope_guard {
+  F f;
+  scope_guard() = delete;
+  scope_guard(const scope_guard &) = delete;
+  scope_guard(scope_guard &&) = default;
+  scope_guard & operator=(const scope_guard &) = delete;
+  scope_guard & operator=(scope_guard &&) = default;
+  scope_guard(F &&f) : f(std::move(f)) {}
+  ~scope_guard() {
```

Comment:
@athanatos could you be a little bit specific on your concern about the exception safety? the `scope_guard` looks good to me and it's way more simpler than the expanded form of its boost alternatives.

i just want to understand if we need to switch over to our home-brew `scope_guard`  at other places where we are using `boost::BOOST_SCOPE_EXIT` or `boost::BOOST_SCOPE_EXIT_ALL` because of your concerns.